src/executor.rs

```diff
@@ -219,34 +216,42 @@ fn invoke_dynamic(
     }) {
         // Process gas requirements and syscall handler.
         match registry.get_type(type_id).unwrap() {
-            CoreTypeConcrete::GasBuiltin(_) => invoke_data.push_aligned(
-                get_integer_layout(128).align(),
-                &[gas as u64, (gas >> 64) as u64],
-            ),
+            CoreTypeConcrete::GasBuiltin(_) => gas.to_bytes(&mut invoke_data),
             CoreTypeConcrete::StarkNet(StarkNetTypeConcrete::System(_)) => {
                 let syscall_handler = syscall_handler
                     .as_mut()
                     .expect("syscall handler is required");
 
-                invoke_data.push_aligned(
-                    get_integer_layout(64).align(),
-                    &[syscall_handler as *mut _ as u64],
-                );
+                (syscall_handler as *mut StarknetSyscallHandlerCallbacks<_>)
+                    .to_bytes(&mut invoke_data);
             }
-            type_info => invoke_data
-                .push(
-                    type_id,
-                    type_info,
-                    if type_info.is_builtin() {
-                        &JitValue::Uint64(0)
-                    } else {
-                        iter.next().unwrap()
-                    },
-                )
-                .unwrap(),
+            type_info if type_info.is_builtin() => 0u64.to_bytes(&mut invoke_data),
+            type_info => JitValueWithInfoWrapper {
+                value: iter.next().unwrap(),
+                type_id,
+                info: type_info,
+
+                arena: &arena,
+                registry,
+            }
+            .to_bytes(&mut invoke_data),
         }
     }
 
+    // Pad invoke data to the 16 byte boundary avoid segfaults.
+    // TODO: Test if the resize must be inserted before or after the data.
```

Comment:
we should create issues to track these TODOs

Reply:
Some (all?) of them are no longer applicable.